File: frontend/src/api/get-recipes-from-ingredients.js
// @flow
import axios from 'axios'
import config from '../config'

export default function getRecipesFromIngredients(ingredients: string[], numResults:number = 5) {
    const ingredQueryString = ingredients.map(i => `i=${i}`).join('&')
    return axios.get(`${config.SERVER_URL}/recipes?${ingredQueryString}`, {
        headers: {
            "Content-Type": "application/json",
        },
        params: {
            r: numResults,
        }
    }).then(res => res.data)
}

File: server/services/scrapers/all_recipe/get_recipes_from_ingredients.py
from time import sleep

from .get_recipe_details import get_recipe_details
from .get_recipe_urls import get_recipe_urls


def get_recipes_from_ingredients(ingredients, num_results=5):
    urls = get_recipe_urls(ingredients)[:num_results]
    recipes = []

    for url in urls:
        recipes.append(get_recipe_details(url))
        sleep(2)

    return recipes
